src/container-instance.class.mts

```diff
@@ -1396,6 +1395,8 @@ export class ContainerInstance implements Disposable {
        */
       // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
       return typeWrapper.extract!(this, constraints ?? ResolutionConstraintFlag.None);
+    } else if (!identifier) {
```

Comment:
Probably simpler just to hoist this to the top, which would still save the potentially unnecessary built-ins check.